### PR TITLE
Fix 1196 handling of Instance value in builder with trailing slash

### PR DESCRIPTION
**Pull request by MarkZuber**
Fixes https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1196

### CODE REVIEW

File: tests/Microsoft.Identity.Test.Unit.net45/AppConfigTests/ConfidentialClientApplicationBuilderTests.cs [hunk 1]
```diff
@@ -40,6 +42,55 @@ public void TestConstructor()
             Assert.AreEqual(null, cca.AppConfig.TenantId);
         }
 
+        private ConfidentialClientApplicationOptions CreateConfidentialClientApplicationOptions()
+        {
+            return new ConfidentialClientApplicationOptions
+            {
+                ClientId = MsalTestConstants.ClientId,
+                ClientSecret = "the_client_secret",
```
Comment (jennyf19) [inline tests/Microsoft.Identity.Test.Unit.net45/AppConfigTests/ConfidentialClientApplicationBuilderTests.cs]: >the_client_secret" [](start = 32, length = 18)

nit: can you add these as constants?

File: src/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs [hunk 2]
```diff
@@ -345,6 +345,7 @@ private string GetDefaultAuthorityInstance()
 
             if (!string.IsNullOrWhiteSpace(Config.Instance))
             {
+                Config.Instance = Config.Instance.TrimEnd(' ', '/');
```
Comment (henrik-me) [inline src/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs]: >Config.Instance = Config.Instance.TrimEnd(' ', '/'); [](start = 16, length = 52)

any reason we are not asking the developer to specify this correctly? exception?

Comment (MarkZuber) [reply src/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs]: Our code later on assumes that this has no / for joining, and when we parse we split on "/".  So this is for us to not screw up.  It's legitimate syntax for the developer to supply the URI as https://login.microsoftonline.com/ or https://login.microsoftoneline.com (no trailing slash).  So this code just makes sure we do the right thing later on when combining these URI paths and parsing them.  

For example, Path.Combine actually solves this for file system paths and does the right thing around making sure there's only one / or \ delimiter.  But new Uri() does NOT do this and so we can end up with https://login.microsoftonline.com//tenantid and then when we parse out the tenant id (by doing a split on '/') we get the empty string for tenant. 

So, bottom line, I don't think we should throw an exception here, we're just keeping values tidy for later processing internally.

Comment (henrik-me) [reply src/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs]: ok. thanks for explaining. 

---
In reply to: [295450204](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1224#discussion_r295450204) [](ancestors = 295450204)
